checker/src/known_names.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use rustc::hir::def_id::DefId;
+use rustc::hir::map::{DefPathData, DisambiguatedDefPathData};
+use rustc::ty::TyCtxt;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// Well known definitions (language provided items) that are treated in special ways.
+#[derive(Serialize, Deserialize, Clone, Copy, Debug, Eq, PartialOrd, PartialEq, Hash, Ord)]
+pub enum KnownNames {
```

Comment:
Plural in a type name indicates a collection. I think this should be called KnowName. But not a blocker.

Reply:
To be fair, an enumeration is a kind of collection. :-)